File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.k = k
        self.result = None

        def inorder(root)-> int:
            if not root or self.result is not None:
                return 
            
            inorder(root.left)

            self.k -=1

            if self.k == 0:
                self.result = root.val
                return
            inorder(root.right)
            
        inorder(root)
        return self.result

File: Data Structures & Algorithms/invert-a-binary-tree/submission-2.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {TreeNode}
     */
    invertTree(root) {
        //BFS
        if(!root) return null;
        let queue = new Queue([root]);

        while(!queue.isEmpty()){
            let node = queue.pop();
            [node.left,node.right] = [node.right,node.left];
            if(node.left) queue.push(node.left);
            if(node.right) queue.push(node.right);
            
        }
        return root;

    }
}
